src/proxy/h2_client.rs

```diff
@@ -258,6 +258,9 @@ impl H2ConnectClient {
         &mut self,
         req: Request<()>,
     ) -> Result<(SendStream<SendBuf>, h2::RecvStream), Error> {
+        // "This function must return `Ready` before `send_request` is called"
+        // We should always be ready though, because we make sure we don't go over the max stream limit out of band.
```

Comment:
Yeah I saw this in the docs at some point but didn't check if we were doing this in the DIY client.

Better to be correct.